File: 03-CPP/08-funciones/03pasoPorValor.cpp
#include <iostream>
using namespace std;

// Definir funcion (Paso por valor: no afecta al valor originals)
void cambiarValor(int parametro)
{
    parametro = 2;
}

int main()
{
    // Ejemplo de paso por valor: Solamente se pasa una copia del valor original.
    int argumento = 10; 

    cout << "Valor antes de funcion " << argumento << endl; 
    cambiarValor(argumento);
    cout << "Valor despues de funcion " << argumento << endl;

    return 0;
}

File: 04-CSHARP/08-Funciones/03pasoPorValor.cs
// Paso por valor en C#

// Definir funcion (Paso por valor: no afecta al valor originals)
void CambiarValor(int parametro)
{
    parametro = 5;
}

// Ejemplo de paso por valor: Solamente se pasa una copia del valor original.
int argumento = 10;

Console.WriteLine($"Valor antes de la funcion: {argumento}");
CambiarValor(argumento);
Console.WriteLine($"Valor despues de la funcion: {argumento}");
